Unable to build without `foundation` support

When you try to build `memory` with `support_foundation` disabled, you get the following error:

```
Building library for memory-0.14.14..
[ 1 of 28] Compiling Data.ByteArray.Types ( Data/ByteArray/Types.hs, dist/build/Data/ByteArray/Types.o )

Data/ByteArray/Types.hs:70:58: error:
    • Variable not in scope:
        memCopy :: Ptr b0 -> Ptr p0 -> Int -> IO ()
    • Perhaps you meant ‘Bytestring.memcpy’ (imported from Data.ByteString.Internal)
   |
70 |     copyByteArrayToPtr a dst = withByteArray a $ \src -> memCopy (castPtr dst) src (length a)
   |    
```

I'm not sure how old this problem is, but it's at least present in `0.14.14` as well as the current git HEAD. It looks like it might be because of misplaced CPP guards for `WITH_FOUNDATION_SUPPORT`. I haven't yet put together a fixed version, but I'd be happy to do so and put in a PR.

--- a/Data/ByteArray/Types.hs
+++ b/Data/ByteArray/Types.hs
@@ -21,6 +21,9 @@ import qualified Data.ByteString.Internal as Bytestring
 import           Foreign.ForeignPtr (withForeignPtr)
 #endif
 
+import           Data.Memory.PtrMethods (memCopy)
+
+
 #ifdef WITH_FOUNDATION_SUPPORT
 
 #if MIN_VERSION_foundation(0,0,14) && MIN_VERSION_basement(0,0,0)
@@ -36,8 +39,6 @@ import qualified Basement.Types.OffsetSize as Base
 import qualified Basement.UArray as Base
 import qualified Basement.String as Base (String, toBytes, Encoding(UTF8))
 import qualified Basement.PrimType as Base (primSizeInBytes)
-
-import           Data.Memory.PtrMethods (memCopy)
 
 #if MIN_VERSION_basement(0,0,5)
 import qualified Basement.UArray.Mutable as BaseMutable (withMutablePtrHint, copyToPtr)

